sbws/core/scanner.py

```diff
@@ -119,7 +118,7 @@ def measure_bandwidth_to_server(session, conf, dest, content_length):
         'target': conf.getfloat('scanner', 'download_target'),
         'max': conf.getfloat('scanner', 'download_max'),
     }
-    while len(results) < num_downloads:
+    while len(results) < num_downloads and not settings.end_event.is_set():
```

Comment:
Hmm, why are we checking the `end_event` here? This does not seem to be part of the `scanner: use end event singleton` commit message explanation.

Reply:
hmm, it does use the ``end_event`` singleton.
The intention is to don't start new request when sbws is ending.

Reply:
Otherwise, finishing sbws would mean to wait for all the new requests.

Reply:
ACK!